lib/lens.ex

```diff
@@ -427,6 +427,35 @@ defmodule Lens do
     lenses |> Enum.reverse() |> Enum.reduce(empty(), &both/2)
   end
 
+  @doc ~S"""
+  Returns a lens that contains what the first lense focuses on, unless it result is empty. In the latter case the
+  result will be what the second lense focuses on.
+
+  This is useful for inserting or updating values.
```

Comment:
Having a motivating example is good, but provide a neutral example first. Probably one with `get_in` would be the easiest to understand.